src/gpl/src/nesterovBase.cpp

```diff
@@ -3003,69 +3053,136 @@ size_t NesterovBaseCommon::createCbkGCell(odb::dbInst* db_inst)
 
 void NesterovBaseCommon::createCbkGNet(odb::dbNet* db_net, bool skip_io_mode)
 {
+  debugPrint(log_, GPL, "callbacks", 2, "NBC createGNet");
   Net gpl_net(db_net, skip_io_mode);
   pb_nets_stor_.push_back(gpl_net);
   GNet gnet(&pb_nets_stor_.back());
   gNetStor_.push_back(gnet);
   GNet* gnet_ptr = &gNetStor_.back();
   gNetMap_[gnet_ptr->net()] = gnet_ptr;
-  db_net_map_[db_net] = gNetStor_.size() - 1;
+  db_net_to_index_map_[db_net] = gNetStor_.size() - 1;
 }
 
 void NesterovBaseCommon::createCbkITerm(odb::dbITerm* iTerm)
 {
+  debugPrint(log_, GPL, "callbacks", 2, "NBC createITerm");
   Pin gpl_pin(iTerm);
   pb_pins_stor_.push_back(gpl_pin);
   GPin gpin(&pb_pins_stor_.back());
   gPinStor_.push_back(gpin);
   GPin* gpin_ptr = &gPinStor_.back();
   gPinMap_[gpin_ptr->pin()] = gpin_ptr;
-  db_iterm_map_[iTerm] = gPinStor_.size() - 1;
+  db_iterm_to_index_map_[iTerm] = gPinStor_.size() - 1;
 }
 
 // assuming fixpointers will be called later
 //  maintaining consistency in NBC::gcellStor_ and NB::gCells_
 void NesterovBase::destroyCbkGCell(odb::dbInst* db_inst)
 {
-  auto db_it = db_inst_index_map_.find(db_inst);
-  if (db_it != db_inst_index_map_.end()) {
+  debugPrint(log_, GPL, "callbacks", 2, "NesterovBase::destroyGCel");
+  auto db_it = db_inst_to_nb_index_map_.find(db_inst);
+  if (db_it != db_inst_to_nb_index_map_.end()) {
     size_t last_index = nb_gcells_.size() - 1;
     size_t gcell_index = db_it->second;
 
     GCellHandle& handle = nb_gcells_[gcell_index];
-    if (handle.isNesterovBaseCommon()) {
-      nbc_->destroyCbkGCell(handle.getIndex());
-    } else {
-      destroyFillerGCell(handle.getIndex());
-    }
+    // TODO delete fillers
+    bool is_filler = handle->isFiller();
 
     if (gcell_index != last_index) {
       std::swap(nb_gcells_[gcell_index], nb_gcells_[last_index]);
     }
     swapAndPopParallelVectors(gcell_index, last_index);
     nb_gcells_.pop_back();
-    db_inst_index_map_.erase(db_it);
+    db_inst_to_nb_index_map_.erase(db_it);
+
+    if (gcell_index != last_index) {
+      if (!nb_gcells_[gcell_index]->isFiller()) {
+        odb::dbInst* swapped_inst
+            = nb_gcells_[gcell_index]->insts()[0]->dbInst();
+        db_inst_to_nb_index_map_.erase(swapped_inst);
+        db_inst_to_nb_index_map_[swapped_inst] = gcell_index;
+      }
+    } else {
+      db_inst_to_nb_index_map_.erase(db_inst);
+    }
```

Comment:
This part of the code is a bit confuse. Could you add some comments on the code for clarity?

I'm not following what are the `gcell_index` and the `last_index` and what they are doing in the `db_inst_to_nb_index_map_` map.